src/Workspaces/Core/Portable/Rename/Renamer.RenameDocumentActionSet.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using System;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    public static partial class Renamer
+    {
+        /// <summary>
+        /// Information about rename document calls that allows them to be applied as individual actions.
+        /// 
+        /// To apply all actions use <see cref="UpdateSolutionAsync(Solution, CancellationToken)"/>, or use a subset
+        /// of the actions by calling <see cref="UpdateSolutionAsync(Solution, ImmutableArray{RenameDocumentAction}, CancellationToken)"/>. 
+        /// Each action has a description of the changes that it will apply that can be presented to a user.
+        /// </summary>
+        public sealed class RenameDocumentActionSet
+        {
+            private readonly DocumentId _documentId;
+            private readonly string _documentName;
+            private readonly IReadOnlyList<string> _documentFolders;
+            private readonly ProjectId _projectId;
+            private readonly OptionSet _optionSet;
```

Comment:
what is this needed for?

Reply:
This is storing the OptionSet passed into the API call, and passed on to any actions that need it. The rename symbol action currently uses it.